plugins/woocommerce/tests/php/src/Blocks/StoreApi/Routes/Checkout.php

```diff
@@ -361,12 +380,20 @@ public function test_checkout_create_account() {
 
 		$customer = get_user_by( 'id', $data['customer_id'] );
 		$this->assertEquals( $customer->user_email, 'testaccount@test.com' );
+
+		// Return WC_Session to original state.
+		WC()->session = $old_session;
 	}
 
 	/**
 	 * Test account creation options.
 	 */
 	public function test_checkout_do_not_create_account() {
+		// We need to replace the WC_Session with a mock because this test relies on cookies being set which
+		// is not easy with PHPUnit. This is a simpler approach.
+		$old_session  = WC()->session;
+		WC()->session = new MockSessionHandler();
```

Comment:
I guess the `woocommerce_session_handler` filter didn't work?

Reply:
Yep, afraid not the only place to set this up was in `plugins/woocommerce/tests/legacy/bootstrap.php` which would affect all tests, not just the ones we need it for. This is a good solution for now.